src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php

```diff
@@ -54,28 +54,28 @@ public function __construct(array $headers = array())
     }
 
     /**
-     * {@inheritdoc}
+     * Returns the headers, with original capitalizations.
+     *
+     * @return array An array of headers
      */
-    public function __toString()
+    public function allPreserveCase()
     {
-        $cookies = '';
-        foreach ($this->getCookies() as $cookie) {
-            $cookies .= 'Set-Cookie: '.$cookie."\r\n";
+        $headers = array();
+        foreach ($this->all() as $name => $value) {
+            $headers[isset($this->headerNames[$name]) ? $this->headerNames[$name] : $name] = $value;
         }
 
-        ksort($this->headerNames);
-
-        return parent::__toString().$cookies;
+        return $headers;
     }
 
-    /**
-     * Returns the headers, with original capitalizations.
-     *
-     * @return array An array of headers
-     */
-    public function allPreserveCase()
+    public function allPreserveCaseWithoutCookies()
     {
-        return array_combine($this->headerNames, $this->headers);
+        $headers = $this->allPreserveCase();
+        if (isset($this->headerNames['set-cookie'])) {
```

Comment:
what if a case is not provided for the ``set-cookie`` header ? you will not remove it at all